common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -92,15 +101,32 @@
                 AuthorizationResult> {
 
     private static final String TAG = MicrosoftStsOAuth2Strategy.class.getSimpleName();
+    private IDevicePopManager mDevicePopManager;
 
     /**
      * Constructor of MicrosoftStsOAuth2Strategy.
      *
-     * @param config MicrosoftStsOAuth2Configuration
+     * @param config  MicrosoftStsOAuth2Configuration
+     * @param options OAuth2StrategyOptions
      */
-    public MicrosoftStsOAuth2Strategy(@NonNull final MicrosoftStsOAuth2Configuration config) {
-        super(config);
-        setTokenEndpoint(config.getTokenEndpoint().toString());
+    public MicrosoftStsOAuth2Strategy(@NonNull final MicrosoftStsOAuth2Configuration config,
+                                      @NonNull final OAuth2StrategyOptions options) {
+        super(config, options);
+        // TODO figure this out!
+        try {
+            mDevicePopManager = new DevicePopManagerImpl(options.getContext());
+
+            if (options.getAuthenticationScheme() instanceof PopAuthenticationSchemeInternal) {
+                // TODO This shouldn't live here.
```

Comment:
This doesn't seem like the correct place to be setting properties on Auth Scheme -- can this happen earlier/elsewhere?